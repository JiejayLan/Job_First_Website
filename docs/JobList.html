<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: JobList</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: JobList</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>JobList<span class="signature">()</span><span class="type-signature"></span></h2>
        
            <div class="class-description"><p>The class jobList will render the page of job list for all the posted job by employer
It will render the job pool containing all the job cards (data from AWS dynamoDB)
It will render the a search and filter section that allows you do search and filter (AWS API)
It will render the a learn more button the job cards so user can direct to jobDescription page</p></div>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    
    <h2>Constructor</h2>
    

    
    <h4 class="name" id="JobList"><span class="type-signature"></span>new JobList<span class="signature">()</span><span class="type-signature"></span></h4>
    

    















<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="pages_jobList.js.html">pages/jobList.js</a>, <a href="pages_jobList.js.html#line29">line 29</a>
    </li></ul></dd>
    

    

    

    
</dl>




















    
    </div>

    

    

    

    

    

    

    
        <h3 class="subsection-title">Members</h3>

        
            
<h4 class="name" id="filterCategory"><span class="type-signature"></span>filterCategory<span class="type-signature"></span></h4>




<div class="description">
    <p>Update the filter object in state with a filter query by job category</p>
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="pages_jobList.js.html">pages/jobList.js</a>, <a href="pages_jobList.js.html#line54">line 54</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="filterDate"><span class="type-signature"></span>filterDate<span class="type-signature"></span></h4>




<div class="description">
    <p>Update the filter object in state with a filter query by post date</p>
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="pages_jobList.js.html">pages/jobList.js</a>, <a href="pages_jobList.js.html#line71">line 71</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="filterEducation"><span class="type-signature"></span>filterEducation<span class="type-signature"></span></h4>




<div class="description">
    <p>Update the filter object in state with a filter query by required education</p>
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="pages_jobList.js.html">pages/jobList.js</a>, <a href="pages_jobList.js.html#line63">line 63</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="filterSalary"><span class="type-signature"></span>filterSalary<span class="type-signature"></span></h4>




<div class="description">
    <p>Update the filter object in state with a filter query by job salary</p>
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="pages_jobList.js.html">pages/jobList.js</a>, <a href="pages_jobList.js.html#line80">line 80</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="filterType"><span class="type-signature"></span>filterType<span class="type-signature"></span></h4>




<div class="description">
    <p>Update the filter object in state with a filter query by job type</p>
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="pages_jobList.js.html">pages/jobList.js</a>, <a href="pages_jobList.js.html#line45">line 45</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="reset"><span class="type-signature"></span>reset<span class="type-signature"></span></h4>




<div class="description">
    <p>Reset the filter object in state back to original value</p>
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="pages_jobList.js.html">pages/jobList.js</a>, <a href="pages_jobList.js.html#line106">line 106</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="searchByName"><span class="type-signature"></span>searchByName<span class="type-signature"></span></h4>




<div class="description">
    <p>Update the filter object in state with a filter query by input name or location</p>
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="pages_jobList.js.html">pages/jobList.js</a>, <a href="pages_jobList.js.html#line98">line 98</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="selectSearch"><span class="type-signature"></span>selectSearch<span class="type-signature"></span></h4>




<div class="description">
    <p>Update the variable searchType with the input value</p>
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="pages_jobList.js.html">pages/jobList.js</a>, <a href="pages_jobList.js.html#line90">line 90</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="setDay"><span class="type-signature"></span>setDay<span class="type-signature"></span></h4>




<div class="description">
    <p>Update the days object in state with an input day</p>
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="pages_jobList.js.html">pages/jobList.js</a>, <a href="pages_jobList.js.html#line122">line 122</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="setSalary"><span class="type-signature"></span>setSalary<span class="type-signature"></span></h4>




<div class="description">
    <p>Update the salary object in state with an input salary</p>
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="pages_jobList.js.html">pages/jobList.js</a>, <a href="pages_jobList.js.html#line114">line 114</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="state"><span class="type-signature"></span>state<span class="type-signature"> :Object</span></h4>




<div class="description">
    <p>the state object to store the query statement for filter and search</p>
</div>



    <h5>Type:</h5>
    <ul>
        <li>
            
<span class="param-type">Object</span>


        </li>
    </ul>





<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="pages_jobList.js.html">pages/jobList.js</a>, <a href="pages_jobList.js.html#line35">line 35</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
    

    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="AddEduForm.html">AddEduForm</a></li><li><a href="AddExpForm.html">AddExpForm</a></li><li><a href="businessProfile.html">businessProfile</a></li><li><a href="Contact.html">Contact</a></li><li><a href="CreateAddressForm.html">CreateAddressForm</a></li><li><a href="JobDescription.html">JobDescription</a></li><li><a href="JobList.html">JobList</a></li><li><a href="ModalForm.html">ModalForm</a></li><li><a href="Profile.html">Profile</a></li><li><a href="UpdateAddressForm.html">UpdateAddressForm</a></li></ul><h3>Global</h3><ul><li><a href="global.html#AppliedJob">AppliedJob</a></li><li><a href="global.html#Education">Education</a></li><li><a href="global.html#Experiences">Experiences</a></li><li><a href="global.html#General">General</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc/jsdoc">JSDoc 3.6.2</a> on Thu May 23 2019 21:05:27 GMT-0400 (Eastern Daylight Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>