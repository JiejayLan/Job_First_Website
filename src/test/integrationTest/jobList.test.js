import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import JobList from '../../pages/jobList';

describe('Job List screenshot test', () => {
    test('Job List page should render correctly', () => {
        const renderer = new ReactShallowRenderer();
        renderer.render(<JobList />);
        expect(renderer.getRenderOutput()).toMatchSnapshot();
    });
});