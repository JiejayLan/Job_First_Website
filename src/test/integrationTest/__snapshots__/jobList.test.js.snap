// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job List screenshot test Job List page should render correctly 1`] = `
<Adapter>
  <Adapter
    style={
      Object {
        "height": "15%",
        "textAlign": "center",
      }
    }
  >
    <Row
      gutter={0}
    >
      <Col
        className="colOne"
        span={8}
      >
        <span
          className="slogon"
        >
          Hire expert for any job, online
        </span>
        <div
          className="slogon2"
        >
          Millisons of small businesses and new immigrants use JobFirst to turn their ideas into reality
        </div>
        <div
          className="buttonGroup"
        >
          <span
            className="slogonButton"
          >
            I want to Hire
          </span>
          <span
            className="slogonButton2"
          >
            I want to Work
          </span>
        </div>
      </Col>
      <Col
        span={12}
        style={
          Object {
            "margin": "-5% 0 -5% 3%",
          }
        }
      >
        <div
          className="banner"
        />
      </Col>
    </Row>
    <div>
      <Group
        compact={true}
      >
        <Select
          choiceTransitionName="zoom"
          defaultValue="Name"
          onChange={[Function]}
          showSearch={false}
          size="large"
          style={
            Object {
              "width": "10%",
            }
          }
          transitionName="slide-up"
        >
          <Option
            value="Name"
          >
            Name
          </Option>
          <Option
            value="Location"
          >
            Location
          </Option>
        </Select>
        <Search
          enterButton="Search"
          id="search"
          onSearch={[Function]}
          placeholder="Search"
          size="large"
          style={
            Object {
              "color": "black",
              "width": "53%",
            }
          }
        />
        <RadioButton
          className="resetButton"
          onClick={[Function]}
          style={
            Object {
              "marginLeft": "1%",
              "width": "6%",
            }
          }
        >
          <Icon
            type="undo"
          />
           Reset
        </RadioButton>
      </Group>
      <Group
        compact={true}
      >
        <div
          style={
            Object {
              "textAlign": "left",
              "width": "30%",
            }
          }
        >
          <h4>
            Filter By Job Category
          </h4>
          <Select
            choiceTransitionName="zoom"
            className="dropBox"
            defaultValue="All"
            filterOption={[Function]}
            onChange={[Function]}
            optionFilterProp="children"
            showSearch={true}
            size="large"
            style={
              Object {
                "width": "100%",
              }
            }
            transitionName="slide-up"
          >
            <Option
              value="Accountancy, banking and finance"
            >
              Accountancy, banking and finance
            </Option>
            <Option
              value="Business, consulting and management"
            >
              Business, consulting and management
            </Option>
            <Option
              value="Charity and voluntary work"
            >
              Charity and voluntary work
            </Option>
            <Option
              value="Creative arts and design"
            >
              Creative arts and design
            </Option>
            <Option
              value="Energy and utilities"
            >
              Energy and utilities
            </Option>
            <Option
              value="Engineering and manufacturing"
            >
              Engineering and manufacturing
            </Option>
            <Option
              value="Environment and agriculture"
            >
              Environment and agriculture
            </Option>
            <Option
              value="Healthcare"
            >
              Healthcare
            </Option>
            <Option
              value="Information technology"
            >
              Information technology
            </Option>
            <Option
              value="Law"
            >
              Law
            </Option>
            <Option
              value="Marketing, advertising and PR"
            >
              Marketing, advertising and PR
            </Option>
            <Option
              value="Media and internet"
            >
              Media and internet
            </Option>
            <Option
              value="Property and construction"
            >
              Property and construction
            </Option>
            <Option
              value="Public services and administration"
            >
              Public services and administration
            </Option>
            <Option
              value="Recruitment and HR"
            >
              Recruitment and HR
            </Option>
            <Option
              value="Retail"
            >
              Retail
            </Option>
            <Option
              value="Sales"
            >
              Sales
            </Option>
            <Option
              value="Science and pharmaceuticals"
            >
              Science and pharmaceuticals
            </Option>
            <Option
              value="Social care"
            >
              Social care
            </Option>
            <Option
              value="Teacher training and education"
            >
              Teacher training and education
            </Option>
            <Option
              value="Transport and logistics"
            >
              Transport and logistics
            </Option>
            <Option
              value="All"
            >
              All
            </Option>
          </Select>
        </div>
        <div
          style={
            Object {
              "textAlign": "left",
              "width": "20%",
            }
          }
        >
          <h4>
            Filter By Job Type
          </h4>
          <Select
            choiceTransitionName="zoom"
            defaultValue="All"
            onChange={[Function]}
            showSearch={false}
            size="large"
            style={
              Object {
                "marginLeft": "2%",
                "width": "98%",
              }
            }
            transitionName="slide-up"
          >
            <Option
              value="Full Time"
            >
              Full Time
            </Option>
            <Option
              value="Part Time"
            >
              Part Time
            </Option>
            <Option
              value="Internship"
            >
              Internship
            </Option>
            <Option
              value="Temporary"
            >
              Temporary
            </Option>
            <Option
              value="All"
            >
              All
            </Option>
          </Select>
        </div>
        <div
          style={
            Object {
              "textAlign": "left",
              "width": "20%",
            }
          }
        >
          <h4>
            Filter By Education Requirement
          </h4>
          <Select
            choiceTransitionName="zoom"
            defaultValue="All"
            onChange={[Function]}
            showSearch={false}
            size="large"
            style={
              Object {
                "marginLeft": "2%",
                "width": "98%",
              }
            }
            transitionName="slide-up"
          >
            <Option
              value="No Requirement"
            >
              No Requirement
            </Option>
            <Option
              value="Associate"
            >
              Associate
            </Option>
            <Option
              value="Bachelor"
            >
              Bachelor
            </Option>
            <Option
              value="Master"
            >
              Master
            </Option>
            <Option
              value="Doctoral"
            >
              Doctoral
            </Option>
            <Option
              value="All"
            >
              All
            </Option>
          </Select>
        </div>
      </Group>
      <Group
        compact={true}
      >
        <div
          style={
            Object {
              "textAlign": "left",
              "width": "20%",
            }
          }
        >
          <h4>
            Minimum Wage
          </h4>
          <Row
            gutter={0}
          >
            <Col
              span={10}
            >
              <Slider
                max={100}
                min={7}
                onChange={[Function]}
                step={0.5}
                tipFormatter={[Function]}
                value={7}
              />
            </Col>
            <Col
              span={1}
            >
              <InputNumber
                max={100}
                min={7}
                onChange={[Function]}
                step={1}
                style={
                  Object {
                    "marginLeft": 16,
                  }
                }
                value={7}
              />
            </Col>
          </Row>
        </div>
        <div
          style={
            Object {
              "textAlign": "center",
              "width": "15%",
            }
          }
        >
          <RadioButton
            className="applyButton"
            onClick={[Function]}
            style={
              Object {
                "marginTop": "5%",
                "width": "40%",
              }
            }
          >
            Apply
          </RadioButton>
        </div>
        <div
          style={
            Object {
              "textAlign": "left",
              "width": "20%",
            }
          }
        >
          <h4>
            Job Posted Within
          </h4>
          <Row
            gutter={0}
          >
            <Col
              span={10}
            >
              <Slider
                max={365}
                min={1}
                onChange={[Function]}
                tipFormatter={[Function]}
                value={365}
              />
            </Col>
            <Col
              span={1}
            >
              <InputNumber
                max={365}
                min={1}
                onChange={[Function]}
                step={10}
                style={
                  Object {
                    "marginLeft": 16,
                  }
                }
                value={365}
              />
            </Col>
          </Row>
        </div>
        <div
          style={
            Object {
              "textAlign": "center",
              "width": "15%",
            }
          }
        >
          <RadioButton
            className="applyButton"
            onClick={[Function]}
            style={
              Object {
                "marginTop": "5%",
                "width": "40%",
              }
            }
          >
            Apply
          </RadioButton>
        </div>
      </Group>
    </div>
  </Adapter>
  <Adapter>
    <Connect
      query={
        Object {
          "query": "query ListPostedJobs(
  $filter: ModelPostedJobFilterInput
  $limit: Int
  $nextToken: String
) {
  listPostedJobs(filter: $filter, limit: $limit, nextToken: $nextToken) {
    items {
      id
      company {
        id
        companyName
        companyEmail
        companyPhone
        companyWebsite
        companyType
        videoURL
        videoPic
        bannerPic
        headquarter
        ceo
        ceoPic
        size
        revenue
        companyPic
        description
        identityID
      }
      jobCategory
      jobTitle
      jobType
      description
      education
      requirements
      salary
      datePosted
      deadline
      location {
        id
        line1
        line2
        city
        postalCode
        state
      }
      applied {
        nextToken
      }
      searchFieldName
      searchFieldLocation
      clickedCounts
    }
    nextToken
  }
}
",
          "variables": Object {
            "filter": Object {
              "clickedCounts": Object {
                "ge": 0,
              },
            },
          },
        }
      }
    >
      [Function]
    </Connect>
  </Adapter>
  <Adapter
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    JobFirst
     ©2019 
    Created by JobFirst Group
  </Adapter>
</Adapter>
`;
